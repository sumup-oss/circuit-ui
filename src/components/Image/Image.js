/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

const baseStyles = () => css`
  label: image;
  display: block;
  height: auto;
  max-height: 100%;
  width: 100%;
`;

const StyledImage = styled('img')(baseStyles);

/**
 * The Image component. Responsive by default.
 */
const Image = props => <StyledImage {...props} />;

Image.propTypes = {
  /**
   * Specifies the source URL of an image
   */
  src: PropTypes.string.isRequired,
  /**
   * Provides alternative information if a user cannot view the image,
   * e.g. because of slow connection, an error in the src attribute, or if the
   * user uses a screen reader.
   */
  alt: PropTypes.string.isRequired
};

/**
 * @component
 */
export default Image;
