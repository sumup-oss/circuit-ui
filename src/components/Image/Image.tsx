/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { HTMLProps } from 'react';
import { css } from '@emotion/core';

import styled from '../../styles/styled';

export interface ImageProps
  extends Omit<HTMLProps<HTMLImageElement>, 'size' | 'crossOrigin'> {
  /**
   * Specifies the source URL of an image
   */
  src: string;
  /**
   * Provides alternative information if a user cannot view the image,
   * e.g. because of slow connection, an error in the src attribute, or if the
   * user uses a screen reader.
   */
  alt: string;
}

const baseStyles = () => css`
  label: image;
  display: block;
  height: auto;
  max-height: 100%;
  width: 100%;
`;

/**
 * The Image component. Responsive by default.
 */
export const Image = styled.img<ImageProps>(baseStyles);
