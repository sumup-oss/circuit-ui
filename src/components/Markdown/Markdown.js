/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import PropTypes from 'prop-types';
import { compiler } from 'markdown-to-jsx';

import deprecate from '../../util/deprecate';

/**
 * Markdown component to parse and render GFM to JSX.
 */
const Markdown = ({
  children,
  overrides,
  forceBlock,
  forceInline,
  transformer
}) => {
  deprecate('Markdown has been deprecated.');

  const options = { overrides, forceBlock, forceInline };
  const transformedMarkdown = transformer(children);
  const html = compiler(transformedMarkdown, options);
  return html;
};

Markdown.propTypes = {
  /**
   * The GitHub flavoured markdown content to be parsed and rendered.
   */
  children: PropTypes.string,
  /**
   * A function to transform/clean the markdown string before it is parsed.
   * Receives the markdown string as first argument and should return a string.
   */
  transformer: PropTypes.func,
  /**
   * Override an HTML tag with a React component, e.g. `a` -> `<Link />`
   * The React component is passed all HTML attributes as props.
   * See https://github.com/probablyup/markdown-to-jsx for documentation.
   */
  overrides: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  /**
   * Treat all markdown strings as "block" elements.
   * See https://github.com/probablyup/markdown-to-jsx for documentation.
   */
  forceBlock: PropTypes.bool,
  /**
   * Treat all markdown strings as "inline" elements.
   * See https://github.com/probablyup/markdown-to-jsx for documentation.
   */
  forceInline: PropTypes.bool
};

Markdown.defaultProps = {
  children: '',
  transformer: md => md,
  overrides: {},
  forceBlock: false,
  forceInline: false
};

/**
 * @component
 */
export default Markdown;
