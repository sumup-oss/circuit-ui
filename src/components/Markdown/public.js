export { default as Markdown } from '.';
