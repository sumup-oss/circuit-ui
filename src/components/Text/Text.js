/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import isPropValid from '@emotion/is-prop-valid';

import { childrenPropType } from '../../util/shared-prop-types';
import { sizes } from '../../styles/constants';

const { KILO, MEGA, GIGA } = sizes;

const mobileSizeMap = {
  [KILO]: KILO,
  [MEGA]: MEGA,
  [GIGA]: MEGA
};

const baseStyles = ({ theme }) => css`
  label: text;
  font-weight: ${theme.fontWeight.regular};
  margin-bottom: ${theme.spacings.mega};
`;

const sizeStyles = ({ theme, size }) => {
  if (!size) {
    return null;
  }

  return css`
    label: ${`text--${size}`};
    font-size: ${theme.typography.text[mobileSizeMap[size]].fontSize};
    line-height: ${theme.typography.text[mobileSizeMap[size]].lineHeight};

    ${theme.mq.kilo} {
      font-size: ${theme.typography.text[size].fontSize};
      line-height: ${theme.typography.text[size].lineHeight};
    }
  `;
};

const boldStyles = ({ theme, bold }) =>
  bold &&
  css`
    label: text--bold;
    font-weight: ${theme.fontWeight.bold};
  `;

const italicStyles = ({ italic }) =>
  italic &&
  css`
    label: text--italic;
    font-style: italic;
  `;

const strikeThroughStyles = ({ strike }) =>
  strike &&
  css`
    label: text--strike-through;
    text-decoration: line-through;
  `;

const marginStyles = ({ noMargin }) =>
  noMargin &&
  css`
    label: text--no-margin;
    margin-bottom: 0;
  `;

export const StyledText = styled('p', {
  shouldForwardProp: prop => isPropValid(prop) && prop !== 'size'
})`
  ${baseStyles};
  ${sizeStyles};
  ${marginStyles};
  ${boldStyles};
  ${italicStyles};
  ${strikeThroughStyles};
`;

/**
 * The Text component is used for long-form text. Typically with
 * <p>, <div>, <article>, or <section> elements. Capable of rendering
 * using different HTML tags.
 */
const Text = props => <StyledText {...props} />;

Text.KILO = KILO;
Text.MEGA = MEGA;
Text.GIGA = GIGA;

Text.propTypes = {
  /**
   * Child nodes to be rendered.
   */
  children: childrenPropType,
  /**
   * A Circuit UI body text size.
   */
  size: PropTypes.oneOf([Text.KILO, Text.MEGA, Text.GIGA]),
  /**
   * Optional additional className string to overwrite styles.
   */
  className: PropTypes.string,
  /**
   * Removes the default bottom margin from the text.
   */
  noMargin: PropTypes.bool,
  /**
   * Bolds the text.
   */
  bold: PropTypes.bool,
  /**
   * Bolds the text.
   */
  italic: PropTypes.bool,
  /**
   * Strikes through the text.
   */
  strike: PropTypes.bool,
  /**
   * The HTML element to render.
   */
  as: PropTypes.string
};

Text.defaultProps = {
  size: Text.MEGA,
  className: '',
  noMargin: false,
  bold: false,
  italic: false,
  children: null
};

/**
 * @component
 */
export default Text;
