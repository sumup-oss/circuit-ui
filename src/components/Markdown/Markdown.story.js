/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';

import Markdown from './Markdown';
import Heading from '../Heading';
import Text from '../Text';
import Image from '../Image';

const content = `# Manu illa amorem committere exit membra minor

## Utraque aut longe ore cruentum est

Lorem markdownum vultusque eligit significent sinistro, et virgis, inter!
Rapuere Apollineam equas coloni urbis conata, *Theti*, colla simul latus magis
et magnis prorumpit flere? Dura quantum.

### Indignantia tamen

Mediis monstravit inmunis Troes non leonis **non** externis cuncta inperfecta
mihi [aristas colligit](http://etluteave.net/) sacros, diu. Se prius
praecordiaque velut totosque norant alipedi candida electae est vestis, posse.

> Adstat Procnes et, in tempora Maenalon moles in nocte relicta, membra nec
> sudore nymphas robur! Adduxit sua equus Cadmeida **egimus** calathis ostendit
> quicquid ad heros. Usus atque, et talisque sum; quin ramis sectamque amore
> texerat Troada. Dedit murmure de iuvenis peraravit, addunt nihil; est est
> ostendens fer cum illa **Galanthida**. Posset aliisque: enim tibi confinia
> terraeque gratus quaque, liquidumque quod: magnum mixta;
> [sicco](http://priusiuno.org/cur.aspx).

### Dumque et membris minas deprimeret Phrygiis sua

Mundus pugno, geniti sic muros Notum tu ad motis; nobis mihi tangi dextra?
**Signans Trinacris errat** attonitoque spicis magistra odioque, aera palus
altum, ense pectora socerumque! Et ipsum vix bis sibi, sitim, aras praemia *qua
quosque*, adde alma. Ait dixit suas gnatis ab facies altera nova **flentemque
teneo**: quoque profusis consequitur delusa et. Consistere adeunt.

### Develat nymphae

Conceperat depositae si manibus utiliter resonare vitat ima mortis ecce. Huic
leni agat pro! Fuit relinque.

1. Quodque nebulas turba orbem aut
2. Nymphae lapsasque
3. Regoque tenere

## Circumstant iam reponere obibat indestrictus sine

Parte quam aequore, nebulas demisere. Iurgia venit finxit nec manibus tamen
cultus coniunx adituque.`;

export default {
  title: 'Typography/Markdown',
  component: Markdown,
  parameters: {
    jest: ['Markdown']
  }
};

export const base = () => <Markdown>{content}</Markdown>;

export const withComponents = () => (
  <Markdown
    overrides={{
      h1: {
        component: Heading,
        props: {
          as: 'h1',
          size: 'zetta'
        }
      },
      h2: {
        component: Heading,
        props: {
          as: 'h2',
          size: 'peta'
        }
      },
      h3: {
        component: Heading,
        props: {
          as: 'h3',
          size: 'giga'
        }
      },
      p: Text,
      img: Image
    }}
  >
    {content}
  </Markdown>
);
