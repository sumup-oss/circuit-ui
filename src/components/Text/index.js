import Text from './Text';

export { StyledText } from './Text';

export default Text;
