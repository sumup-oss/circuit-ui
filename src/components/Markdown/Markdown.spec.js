/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';

import Markdown from '.';
import Image from '../Image';
import Text from '../Text';

const markdown = `# Manu illa amorem committere exit membra minor

## Utraque aut longe ore cruentum est

Lorem markdownum vultusque eligit significent sinistro, et virgis, inter!
Rapuere Apollineam equas coloni urbis conata, *Theti*, colla simul latus magis
et magnis prorumpit flere? Dura quantum.

### Indignantia tamen

Mediis monstravit inmunis Troes non leonis **non** externis cuncta inperfecta
mihi [aristas colligit](http://etluteave.net/) sacros, diu. Se prius
praecordiaque velut totosque norant alipedi candida electae est vestis, posse.

> Adstat Procnes et, in tempora Maenalon moles in nocte relicta, membra nec
> sudore nymphas robur! Adduxit sua equus Cadmeida **egimus** calathis ostendit
> quicquid ad heros. Usus atque, et talisque sum; quin ramis sectamque amore
> texerat Troada. Dedit murmure de iuvenis peraravit, addunt nihil; est est
> ostendens fer cum illa **Galanthida**. Posset aliisque: enim tibi confinia
> terraeque gratus quaque, liquidumque quod: magnum mixta;
> [sicco](http://priusiuno.org/cur.aspx).

### Dumque et membris minas deprimeret Phrygiis sua

Mundus pugno, geniti sic muros Notum tu ad motis; nobis mihi tangi dextra?
**Signans Trinacris errat** attonitoque spicis magistra odioque, aera palus
altum, ense pectora socerumque! Et ipsum vix bis sibi, sitim, aras praemia *qua
quosque*, adde alma. Ait dixit suas gnatis ab facies altera nova **flentemque
teneo**: quoque profusis consequitur delusa et. Consistere adeunt.

### Develat nymphae

Conceperat depositae si manibus utiliter resonare vitat ima mortis ecce. Huic
leni agat pro! Fuit relinque.

1. Quodque nebulas turba orbem aut
2. Nymphae lapsasque
3. Regoque tenere

## Circumstant iam reponere obibat indestrictus sine

![A random cute puppy](http://www.placepuppy.net/800/500)

Parte quam aequore, nebulas demisere. Iurgia venit finxit nec manibus tamen
cultus coniunx adituque.`;

describe('Markdown', () => {
  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<Markdown>{markdown}</Markdown>);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });

  /**
   * Logic tests.
   */
  it('should parse and render Markdown to HTML', () => {
    const actual = create(<Markdown>{markdown}</Markdown>);
    expect(actual).toMatchSnapshot();
  });

  it('should transform the Markdown before parsing it', () => {
    const actual = create(
      <Markdown transformer={md => md.replace(/#/g, '##')}>{markdown}</Markdown>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should override HTML tags with React components', () => {
    const actual = create(
      <Markdown overrides={{ img: Image, p: Text }}>{markdown}</Markdown>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should treat all markdown strings as "block" elements.', () => {
    const actual = create(<Markdown forceBlock>{markdown}</Markdown>);
    expect(actual).toMatchSnapshot();
  });

  it('should treat all markdown strings as "inline" elements.', () => {
    const actual = create(<Markdown forceInline>{markdown}</Markdown>);
    expect(actual).toMatchSnapshot();
  });
});
