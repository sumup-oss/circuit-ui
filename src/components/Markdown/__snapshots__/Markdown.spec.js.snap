// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown should override HTML tags with React components 1`] = `
.circuit-0 {
  font-weight: 400;
  margin-bottom: 16px;
  font-size: 16px;
  line-height: 24px;
}

@media (min-width:480px) {
  .circuit-0 {
    font-size: 16px;
    line-height: 24px;
  }
}

.circuit-10 {
  display: block;
  height: auto;
  max-height: 100%;
  width: 100%;
}

<div>
  <h1
    id="manu-illa-amorem-committere-exit-membra-minor"
  >
    Manu illa amorem committere exit membra minor
  </h1>
  <h2
    id="utraque-aut-longe-ore-cruentum-est"
  >
    Utraque aut longe ore cruentum est
  </h2>
  <p
    class=" circuit-0 circuit-1"
  >
    Lorem markdownum vultusque eligit significent sinistro, et virgis, inter!
Rapuere Apollineam equas coloni urbis conata, 
    <em>
      Theti
    </em>
    , colla simul latus magis
et magnis prorumpit flere? Dura quantum.
  </p>
  <h3
    id="indignantia-tamen"
  >
    Indignantia tamen
  </h3>
  <p
    class=" circuit-0 circuit-1"
  >
    Mediis monstravit inmunis Troes non leonis 
    <strong>
      non
    </strong>
     externis cuncta inperfecta
mihi 
    <a
      href="http://etluteave.net/"
    >
      aristas colligit
    </a>
     sacros, diu. Se prius
praecordiaque velut totosque norant alipedi candida electae est vestis, posse.
  </p>
  <blockquote>
    <p
      class=" circuit-0 circuit-1"
    >
      Adstat Procnes et, in tempora Maenalon moles in nocte relicta, membra nec
sudore nymphas robur! Adduxit sua equus Cadmeida 
      <strong>
        egimus
      </strong>
       calathis ostendit
quicquid ad heros. Usus atque, et talisque sum; quin ramis sectamque amore
texerat Troada. Dedit murmure de iuvenis peraravit, addunt nihil; est est
ostendens fer cum illa 
      <strong>
        Galanthida
      </strong>
      . Posset aliisque: enim tibi confinia
terraeque gratus quaque, liquidumque quod: magnum mixta;

      <a
        href="http://priusiuno.org/cur.aspx"
      >
        sicco
      </a>
      .
    </p>
  </blockquote>
  <h3
    id="dumque-et-membris-minas-deprimeret-phrygiis-sua"
  >
    Dumque et membris minas deprimeret Phrygiis sua
  </h3>
  <p
    class=" circuit-0 circuit-1"
  >
    Mundus pugno, geniti sic muros Notum tu ad motis; nobis mihi tangi dextra?

    <strong>
      Signans Trinacris errat
    </strong>
     attonitoque spicis magistra odioque, aera palus
altum, ense pectora socerumque! Et ipsum vix bis sibi, sitim, aras praemia *qua
quosque
    <em>
      , adde alma. Ait dixit suas gnatis ab facies altera nova *
    </em>
    flentemque
teneo
    <em />
    : quoque profusis consequitur delusa et. Consistere adeunt.
  </p>
  <h3
    id="develat-nymphae"
  >
    Develat nymphae
  </h3>
  <p
    class=" circuit-0 circuit-1"
  >
    Conceperat depositae si manibus utiliter resonare vitat ima mortis ecce. Huic
leni agat pro! Fuit relinque.
  </p>
  <ol
    start="1"
  >
    <li>
      Quodque nebulas turba orbem aut
    </li>
    <li>
      Nymphae lapsasque
    </li>
    <li>
      Regoque tenere
    </li>
  </ol>
  <h2
    id="circumstant-iam-reponere-obibat-indestrictus-sine"
  >
    Circumstant iam reponere obibat indestrictus sine
  </h2>
  <p
    class=" circuit-0 circuit-1"
  >
    <img
      alt="A random cute puppy"
      class="circuit-10 circuit-11"
      src="http://www.placepuppy.net/800/500"
    />
  </p>
  <p
    class=" circuit-0 circuit-1"
  >
    Parte quam aequore, nebulas demisere. Iurgia venit finxit nec manibus tamen
cultus coniunx adituque.
  </p>
</div>
`;

exports[`Markdown should parse and render Markdown to HTML 1`] = `
<div>
  <h1
    id="manu-illa-amorem-committere-exit-membra-minor"
  >
    Manu illa amorem committere exit membra minor
  </h1>
  <h2
    id="utraque-aut-longe-ore-cruentum-est"
  >
    Utraque aut longe ore cruentum est
  </h2>
  <p>
    Lorem markdownum vultusque eligit significent sinistro, et virgis, inter!
Rapuere Apollineam equas coloni urbis conata, 
    <em>
      Theti
    </em>
    , colla simul latus magis
et magnis prorumpit flere? Dura quantum.
  </p>
  <h3
    id="indignantia-tamen"
  >
    Indignantia tamen
  </h3>
  <p>
    Mediis monstravit inmunis Troes non leonis 
    <strong>
      non
    </strong>
     externis cuncta inperfecta
mihi 
    <a
      href="http://etluteave.net/"
    >
      aristas colligit
    </a>
     sacros, diu. Se prius
praecordiaque velut totosque norant alipedi candida electae est vestis, posse.
  </p>
  <blockquote>
    <p>
      Adstat Procnes et, in tempora Maenalon moles in nocte relicta, membra nec
sudore nymphas robur! Adduxit sua equus Cadmeida 
      <strong>
        egimus
      </strong>
       calathis ostendit
quicquid ad heros. Usus atque, et talisque sum; quin ramis sectamque amore
texerat Troada. Dedit murmure de iuvenis peraravit, addunt nihil; est est
ostendens fer cum illa 
      <strong>
        Galanthida
      </strong>
      . Posset aliisque: enim tibi confinia
terraeque gratus quaque, liquidumque quod: magnum mixta;

      <a
        href="http://priusiuno.org/cur.aspx"
      >
        sicco
      </a>
      .
    </p>
  </blockquote>
  <h3
    id="dumque-et-membris-minas-deprimeret-phrygiis-sua"
  >
    Dumque et membris minas deprimeret Phrygiis sua
  </h3>
  <p>
    Mundus pugno, geniti sic muros Notum tu ad motis; nobis mihi tangi dextra?

    <strong>
      Signans Trinacris errat
    </strong>
     attonitoque spicis magistra odioque, aera palus
altum, ense pectora socerumque! Et ipsum vix bis sibi, sitim, aras praemia *qua
quosque
    <em>
      , adde alma. Ait dixit suas gnatis ab facies altera nova *
    </em>
    flentemque
teneo
    <em />
    : quoque profusis consequitur delusa et. Consistere adeunt.
  </p>
  <h3
    id="develat-nymphae"
  >
    Develat nymphae
  </h3>
  <p>
    Conceperat depositae si manibus utiliter resonare vitat ima mortis ecce. Huic
leni agat pro! Fuit relinque.
  </p>
  <ol
    start="1"
  >
    <li>
      Quodque nebulas turba orbem aut
    </li>
    <li>
      Nymphae lapsasque
    </li>
    <li>
      Regoque tenere
    </li>
  </ol>
  <h2
    id="circumstant-iam-reponere-obibat-indestrictus-sine"
  >
    Circumstant iam reponere obibat indestrictus sine
  </h2>
  <p>
    <img
      alt="A random cute puppy"
      src="http://www.placepuppy.net/800/500"
    />
  </p>
  <p>
    Parte quam aequore, nebulas demisere. Iurgia venit finxit nec manibus tamen
cultus coniunx adituque.
  </p>
</div>
`;

exports[`Markdown should transform the Markdown before parsing it 1`] = `
<div>
  <h2
    id="manu-illa-amorem-committere-exit-membra-minor"
  >
    Manu illa amorem committere exit membra minor
  </h2>
  <h4
    id="utraque-aut-longe-ore-cruentum-est"
  >
    Utraque aut longe ore cruentum est
  </h4>
  <p>
    Lorem markdownum vultusque eligit significent sinistro, et virgis, inter!
Rapuere Apollineam equas coloni urbis conata, 
    <em>
      Theti
    </em>
    , colla simul latus magis
et magnis prorumpit flere? Dura quantum.
  </p>
  <h6
    id="indignantia-tamen"
  >
    Indignantia tamen
  </h6>
  <p>
    Mediis monstravit inmunis Troes non leonis 
    <strong>
      non
    </strong>
     externis cuncta inperfecta
mihi 
    <a
      href="http://etluteave.net/"
    >
      aristas colligit
    </a>
     sacros, diu. Se prius
praecordiaque velut totosque norant alipedi candida electae est vestis, posse.
  </p>
  <blockquote>
    <p>
      Adstat Procnes et, in tempora Maenalon moles in nocte relicta, membra nec
sudore nymphas robur! Adduxit sua equus Cadmeida 
      <strong>
        egimus
      </strong>
       calathis ostendit
quicquid ad heros. Usus atque, et talisque sum; quin ramis sectamque amore
texerat Troada. Dedit murmure de iuvenis peraravit, addunt nihil; est est
ostendens fer cum illa 
      <strong>
        Galanthida
      </strong>
      . Posset aliisque: enim tibi confinia
terraeque gratus quaque, liquidumque quod: magnum mixta;

      <a
        href="http://priusiuno.org/cur.aspx"
      >
        sicco
      </a>
      .
    </p>
  </blockquote>
  <h6
    id="dumque-et-membris-minas-deprimeret-phrygiis-sua"
  >
    Dumque et membris minas deprimeret Phrygiis sua
  </h6>
  <p>
    Mundus pugno, geniti sic muros Notum tu ad motis; nobis mihi tangi dextra?

    <strong>
      Signans Trinacris errat
    </strong>
     attonitoque spicis magistra odioque, aera palus
altum, ense pectora socerumque! Et ipsum vix bis sibi, sitim, aras praemia *qua
quosque
    <em>
      , adde alma. Ait dixit suas gnatis ab facies altera nova *
    </em>
    flentemque
teneo
    <em />
    : quoque profusis consequitur delusa et. Consistere adeunt.
  </p>
  <h6
    id="develat-nymphae"
  >
    Develat nymphae
  </h6>
  <p>
    Conceperat depositae si manibus utiliter resonare vitat ima mortis ecce. Huic
leni agat pro! Fuit relinque.
  </p>
  <ol
    start="1"
  >
    <li>
      Quodque nebulas turba orbem aut
    </li>
    <li>
      Nymphae lapsasque
    </li>
    <li>
      Regoque tenere
    </li>
  </ol>
  <h4
    id="circumstant-iam-reponere-obibat-indestrictus-sine"
  >
    Circumstant iam reponere obibat indestrictus sine
  </h4>
  <p>
    <img
      alt="A random cute puppy"
      src="http://www.placepuppy.net/800/500"
    />
  </p>
  <p>
    Parte quam aequore, nebulas demisere. Iurgia venit finxit nec manibus tamen
cultus coniunx adituque.
  </p>
</div>
`;

exports[`Markdown should treat all markdown strings as "block" elements. 1`] = `
<div>
  <h1
    id="manu-illa-amorem-committere-exit-membra-minor"
  >
    Manu illa amorem committere exit membra minor
  </h1>
  <h2
    id="utraque-aut-longe-ore-cruentum-est"
  >
    Utraque aut longe ore cruentum est
  </h2>
  <p>
    Lorem markdownum vultusque eligit significent sinistro, et virgis, inter!
Rapuere Apollineam equas coloni urbis conata, 
    <em>
      Theti
    </em>
    , colla simul latus magis
et magnis prorumpit flere? Dura quantum.
  </p>
  <h3
    id="indignantia-tamen"
  >
    Indignantia tamen
  </h3>
  <p>
    Mediis monstravit inmunis Troes non leonis 
    <strong>
      non
    </strong>
     externis cuncta inperfecta
mihi 
    <a
      href="http://etluteave.net/"
    >
      aristas colligit
    </a>
     sacros, diu. Se prius
praecordiaque velut totosque norant alipedi candida electae est vestis, posse.
  </p>
  <blockquote>
    <p>
      Adstat Procnes et, in tempora Maenalon moles in nocte relicta, membra nec
sudore nymphas robur! Adduxit sua equus Cadmeida 
      <strong>
        egimus
      </strong>
       calathis ostendit
quicquid ad heros. Usus atque, et talisque sum; quin ramis sectamque amore
texerat Troada. Dedit murmure de iuvenis peraravit, addunt nihil; est est
ostendens fer cum illa 
      <strong>
        Galanthida
      </strong>
      . Posset aliisque: enim tibi confinia
terraeque gratus quaque, liquidumque quod: magnum mixta;

      <a
        href="http://priusiuno.org/cur.aspx"
      >
        sicco
      </a>
      .
    </p>
  </blockquote>
  <h3
    id="dumque-et-membris-minas-deprimeret-phrygiis-sua"
  >
    Dumque et membris minas deprimeret Phrygiis sua
  </h3>
  <p>
    Mundus pugno, geniti sic muros Notum tu ad motis; nobis mihi tangi dextra?

    <strong>
      Signans Trinacris errat
    </strong>
     attonitoque spicis magistra odioque, aera palus
altum, ense pectora socerumque! Et ipsum vix bis sibi, sitim, aras praemia *qua
quosque
    <em>
      , adde alma. Ait dixit suas gnatis ab facies altera nova *
    </em>
    flentemque
teneo
    <em />
    : quoque profusis consequitur delusa et. Consistere adeunt.
  </p>
  <h3
    id="develat-nymphae"
  >
    Develat nymphae
  </h3>
  <p>
    Conceperat depositae si manibus utiliter resonare vitat ima mortis ecce. Huic
leni agat pro! Fuit relinque.
  </p>
  <ol
    start="1"
  >
    <li>
      Quodque nebulas turba orbem aut
    </li>
    <li>
      Nymphae lapsasque
    </li>
    <li>
      Regoque tenere
    </li>
  </ol>
  <h2
    id="circumstant-iam-reponere-obibat-indestrictus-sine"
  >
    Circumstant iam reponere obibat indestrictus sine
  </h2>
  <p>
    <img
      alt="A random cute puppy"
      src="http://www.placepuppy.net/800/500"
    />
  </p>
  <p>
    Parte quam aequore, nebulas demisere. Iurgia venit finxit nec manibus tamen
cultus coniunx adituque.
  </p>
</div>
`;

exports[`Markdown should treat all markdown strings as "inline" elements. 1`] = `
<span>
  # Manu illa amorem committere exit membra minor

## Utraque aut longe ore cruentum est

Lorem markdownum vultusque eligit significent sinistro, et virgis, inter!
Rapuere Apollineam equas coloni urbis conata, 
  <em>
    Theti
  </em>
  , colla simul latus magis
et magnis prorumpit flere? Dura quantum.

### Indignantia tamen

Mediis monstravit inmunis Troes non leonis 
  <strong>
    non
  </strong>
   externis cuncta inperfecta
mihi 
  <a
    href="http://etluteave.net/"
  >
    aristas colligit
  </a>
   sacros, diu. Se prius
praecordiaque velut totosque norant alipedi candida electae est vestis, posse.

&gt; Adstat Procnes et, in tempora Maenalon moles in nocte relicta, membra nec
&gt; sudore nymphas robur! Adduxit sua equus Cadmeida 
  <strong>
    egimus
  </strong>
   calathis ostendit
&gt; quicquid ad heros. Usus atque, et talisque sum; quin ramis sectamque amore
&gt; texerat Troada. Dedit murmure de iuvenis peraravit, addunt nihil; est est
&gt; ostendens fer cum illa 
  <strong>
    Galanthida
  </strong>
  . Posset aliisque: enim tibi confinia
&gt; terraeque gratus quaque, liquidumque quod: magnum mixta;
&gt; 
  <a
    href="http://priusiuno.org/cur.aspx"
  >
    sicco
  </a>
  .

### Dumque et membris minas deprimeret Phrygiis sua

Mundus pugno, geniti sic muros Notum tu ad motis; nobis mihi tangi dextra?

  <strong>
    Signans Trinacris errat
  </strong>
   attonitoque spicis magistra odioque, aera palus
altum, ense pectora socerumque! Et ipsum vix bis sibi, sitim, aras praemia *qua
quosque
  <em>
    , adde alma. Ait dixit suas gnatis ab facies altera nova *
  </em>
  flentemque
teneo
  <em />
  : quoque profusis consequitur delusa et. Consistere adeunt.

### Develat nymphae

Conceperat depositae si manibus utiliter resonare vitat ima mortis ecce. Huic
leni agat pro! Fuit relinque.

1. Quodque nebulas turba orbem aut
2. Nymphae lapsasque
3. Regoque tenere

## Circumstant iam reponere obibat indestrictus sine


  <img
    alt="A random cute puppy"
    src="http://www.placepuppy.net/800/500"
  />
  

Parte quam aequore, nebulas demisere. Iurgia venit finxit nec manibus tamen
cultus coniunx adituque.
</span>
`;
